libvcs4j-mapping/src/main/java/de/unibremen/informatik/st/libvcs4j/mapping/Mapping.java

```diff
@@ -187,6 +230,346 @@ public int getOrdinal() {
 	public Result<T> map(final Collection<Mappable<T>> from,
 			final Collection<Mappable<T>> to, final RevisionRange range)
 			throws NullPointerException, IOException {
-		throw new UnsupportedOperationException("Not yet implemented");
+		Validate.notNull(from);
```

Comment:
You should use [Validate.noNullElements](https://github.com/uni-bremen-agst/libvcs4j/blob/master/libvcs4j-api/src/main/java/de/unibremen/informatik/st/libvcs4j/Validate.java#L73) to validate that neither `from` nor `to` contains null values.